### PR TITLE
Add spec to get the block size of gfs2 file system

### PR DESCRIPTION
**Pull request by huali027**
Signed-off-by: Huanhuan Li <huali@redhat.com>

### CODE REVIEW

File: insights/parsers/tests/test_gfs2_file_system_block_size.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+BLOCK_SIZE_OUTPUT_2 = """
+abc
+""".strip()
+
+
+def test_exp():
+    with pytest.raises(SkipException):
+        gfs2_file_system_block_size.GFS2FileSystemBlockSize(context_wrap(BLOCK_SIZE_OUTPUT_2))
+
+
```
Comment (bfahr) [inline insights/parsers/tests/test_gfs2_file_system_block_size.py:20]: @huali027 you should add some additional testing for the parser here, and not rely solely on the documentation test and negative test.

**Review by bfahr — CHANGES_REQUESTED** [review]

File: insights/parsers/tests/test_gfs2_file_system_block_size.py (hunk 1 above)
Comment (huali027) [reply insights/parsers/tests/test_gfs2_file_system_block_size.py:20]: @bfahr, I thought it was enough to test right value and bad value. Since the output is every simple. Let me check to add more test, thanks for the review.

File: insights/specs/default.py [hunk 2]
```diff
@@ -383,6 +385,30 @@ def du_dirs_list(broker):
...
+        systems.
+        And only run the ``stat`` command on RHEL version that's less than
+        8.3. With 8.3 and later, the command ``blkid`` will also output the
+        block size info.
+
+        Returns:
+            list: a list of mount points of which the file system type is gfs2
+        """
+        gfs2_mount_points = []
+        if (broker[RedHatRelease] and
```
Comment (xiangce) [inline insights/specs/default.py:402]: The Components IsRhel* is just enhanced and ".minor" is added, would you please change to use the IsRhel* here?

Comment (huali027) [reply insights/specs/default.py:402]: @xiangce, sure.

File: insights/specs/default.py [hunk 3]
```diff
@@ -382,6 +383,29 @@ def du_dirs_list(broker):
...
+        gfs2_mount_points = []
+        if (broker.get(IsRhel6) or broker.get(IsRhel7) or
+                (broker.get(IsRhel8) and broker[IsRhel8].minor < 3)):
+            for mnt in broker[Mount]:
+                if mnt.mount_type == "gfs2":
+                    gfs2_mount_points.append(mnt.mount_point)
+        if gfs2_mount_points:
+            return gfs2_mount_points
+        raise SkipComponent
+    gfs2_file_system_block_size = foreach_execute(gfs2_mount_points, "/usr/bin/stat -fc %%s %s")
```
Comment (bfahr) [inline insights/specs/default.py:408]: @huali027 you also need to add a spec to `insights_archive.py` to enable access to the data in the archive.  I believe this will be the glob  `insights_commands\stat_-fc_s*`.

**Review by bfahr — CHANGES_REQUESTED** [review]

File: insights/specs/default.py (hunk 3 above)
Comment (huali027) [reply insights/specs/default.py:408]: @bfahr, since it is run by "foreach_execute", all the results are saved to "gfs2_file_system_block_size" and I tested it too, it works well. Do we still need to use glob to add a spec in insights_archive.py explicitly?

Comment (xiangce) [reply insights/specs/default.py:408]: @bfahr - this spec depends on a @datasource which is not supported by the old collection, which means it cannot be collected by the old collection. And the new collection will set up the relationship by generating a JSON file to the `meta_data` dir for all specs. Is it necessary to add it to the `insights_archive.py`?

Comment (bfahr) [reply insights/specs/default.py:408]: @huali027 @xiangce you are correct.  I had tested it incorrectly, and when I reran some additional tests I see that it works correctly as it.

**Review by bfahr — APPROVED** [review]
